test/dotnet-uninstall.Tests/TestUtils/EqualityComparisonTestUtils.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System;
+using FluentAssertions;
+
+namespace Microsoft.DotNet.Tools.Uninstall.Tests.TestUtils
+{
+    public static class EqualityComparisonTestUtils<T>
```

Comment:
[out of topic] my previous teammate Jay wrote a library that is similar to this (although it is not on dotnet core) http://jbazuzicode.blogspot.com/2016/06/valuetypeassertions.html

it is interesting to look at the implementation https://github.com/JayBazuzi/ValueTypeAssertions/blob/master/ValueTypeAssertions/ValueTypeAssertions.cs

Reply:
it is quite some work to write WholeValue in C#

Reply:
Yes I agree. Figuring out dynamic dispatch for C# is also especially hard.

Thanks for your links! Worth reading.